File: userService.ts
interface User {
  id: number;
  name: string;
  email: string;
}

class UserService {
  private users: Map<number, User> = new Map();

  createUser(id: number, name: string, email: string): User {
    if (this.users.has(id)) {
      throw new Error("User exists");
    }

    const user: User = { id, name, email };
    this.users.set(id, user);
    return user;
  }

  getUser(id: number): User {
    const user = this.users.get(id);
    if (!user) throw new Error("Not found");
    return user;
  }

  listUsers(): User[] {
    return Array.from(this.users.values());
  }
}

const service = new UserService();
service.createUser(1, "Alice", "a@mail.com");
console.log(service.listUsers());

File: UserService.java
import java.util.*;

class User {
    int id;
    String name;
    String email;

    User(int id, String name, String email) {
        this.id = id;
        this.name = name;
        this.email = email;
    }
}

class UserService {
    private Map<Integer, User> users = new HashMap<>();

    public void createUser(int id, String name, String email) {
        users.put(id, new User(id, name, email));
    }

    public User getUser(int id) {
        return users.get(id);
    }

    public List<User> listUsers() {
        return new ArrayList<>(users.values());
    }
}

public class Main {
    public static void main(String[] args) {
        UserService service = new UserService();
        service.createUser(1, "Alice", "a@mail.com");
        service.createUser(2, "Bob", "b@mail.com");

        for (User u : service.listUsers()) {
            System.out.println(u.name);
        }
    }
}
